src/include/migraphx/process.hpp

```diff
@@ -50,8 +52,14 @@ struct MIGRAPHX_EXPORT process
 
     process& cwd(const fs::path& p);
 
-    void exec();
-    void write(std::function<void(process::writer)> pipe_in);
+    void exec(std::string_view args = "", std::string_view envs = "");
+    void write(std::function<void(process::writer)> pipe_in,
+               std::string_view args = "",
+               std::string_view envs = "");
+
+#ifdef _WIN32
+    process& redirect_std_out(void* rout);
```

Comment:
I think this should be `process& read(const process::writer& pipe_out)`. For consistency, it would be good to change `write` to `process& write(std::function<void(process::writer)> pipe_in)` and then require `exec` to run the process instead of having `write` call `exec`.